### ISSUE TITLE
Successfully built the libraries, but I can't get cmake to find them

### ISSUE THREAD

**Issue by PetrKryslUCSD** [xianyi/OpenBLAS#1522]
How do I do that?  I found a hint to specify
```
set(BLAS_DIR /path/to/blas)
find_package(BLAS REQUIRED)
```
But what is the correct  `/path/to/blas`? I built in `OpenBLAS/build`. However specifying that folder does not work.

**Comment by martin-frbg** [comment]
From reading the cmake documentation for FindBLAS, try setting BLAS_LIBS rather than BLAS_DIR. You may also want to run `make install` in the OpenBLAS build directory first to put it somewhere sane (usually either /usr/local or /opt by default, depending on your system), which also rebuilds and renames a couple of header files from the build directory during the copy.

**Comment by PetrKryslUCSD** [comment]
Thank you very much for getting back to me.
Sorry: should have been more explicit about my set up. I do this on Windows 10. Also, I have used cmake to build. Now I ran `cmake --install ..` and the resulting build looks like this:
![image](https://user-images.githubusercontent.com/9855832/38770883-713d8f24-3fce-11e8-8061-09f542e660db.png)
What exactly is the directory I want to set BLAS_LIBS to?
Thanks.

**Comment by martin-frbg** [comment]
Did not see your other issue before I replied to this one. I am actually not quite familiar with the windows builds, so my lame answer has to be "whereever you see an openblas.dll or openblas.a" - from my reading of the FindBLAS procedure, it tries to find a library in the given location and checks if it sees any GEMM function in it. Possibly you will have to copy or rename the "openblas" dll to just "blas" for it to get found.

**Comment by martin-frbg** [comment]
Seems from cmake 3.6.0 onwards FindBLAS (and FindLAPACK) knows to recognize OpenBLAS without renaming. With your conda setup you should be at cmake 3.10.0 or 3.11.0 already.

**Comment by PetrKryslUCSD** [comment]
I don't know, I get
```
$ cmake ..
CMake Error at C:/Program Files/CMake/share/cmake-3.11/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
  Could NOT find BLAS (missing: BLAS_LIBRARIES)
Call Stack (most recent call first):
  C:/Program Files/CMake/share/cmake-3.11/Modules/FindPackageHandleStandardArgs.cmake:378 (_FPHSA_FAILURE_MESSAGE)
  C:/Program Files/CMake/share/cmake-3.11/Modules/FindBLAS.cmake:705 (find_package_handle_standard_args)
  CMakeLists.txt:11 (find_package)
```
for
```
if(USE_BLAS)
  set(BLAS_DIR "C:\\Users\\PetrKrysl\\Documents\\Work-in-progress\\OpenBLAS\\build\\lib\\RELEASE")
  find_package(BLAS REQUIRED)
  find_package(LAPACK REQUIRED)
  add_definitions(-DUSE_BLAS)
endif()
```
in CMakeLists.txt.
In the folder `C:\\Users\\PetrKrysl\\Documents\\Work-in-progress\\OpenBLAS\\build\\lib\\RELEASE` I have
```
$ ls C:\\Users\\PetrKrysl\\Documents\\Work-in-progress\\OpenBLAS\\build\\lib\\RELEASE
libopenblas.dll*  libopenblas.exp  libopenblas.lib  libopenblas.pdb
```

**Comment by martin-frbg** [comment]
Did you try setting BLAS_LIBS instead of BLAS_DIR ?

**Comment by PetrKryslUCSD** [comment]
Same result with
```
if(USE_BLAS)
  set(BLAS_LIBS "C:\\Users\\PetrKrysl\\Documents\\Work-in-progress\\OpenBLAS\\build\\lib\\RELEASE")
  find_package(BLAS REQUIRED)
  find_package(LAPACK REQUIRED)
  add_definitions(-DUSE_BLAS)
endif()
```

**Comment by martin-frbg** [comment]
From https://stackoverflow.com/questions/20328991/can-cmake-findblas-find-openblas it appears you may have to call cmake with `-DBLA_VENDOR=OpenBLAS` and/or set -DCMAKE_PREFIX_PATH to the location of the library.

**Comment by martin-frbg** [comment]
From the actual FindBLAS.cmake (assuming I grabbed the correct version):
1. -DBLA_VENDOR is optional, if not set find_blas will try all the implementations it knows about
2. BLAS_DIR, if set, is expected to be the base directory of the blas implementation,
BLAS_LIBDIR is the path containing the library file itself. 
3. If neither BLAS_DIR nor BLAS_LIBDIR is set, the "usual" environment variables like LD_LIBRARY_PATH and the "usual" library locations of the operating system are searched

**Comment by isuruf** [comment]
You need to install OpenBLAS by doing,
```
cmake --build . --config Release --target install
```
and then the openblas libraries  and headers will be installed into a directory which you'll see from the logs. (Something like C:\Program Files\openblas).

Then set BLAS_DIR in your CMakeLists.txt to that directory.

**Comment by martin-frbg** [comment]
Petr, did you get this sorted ?

**Comment by PetrKryslUCSD** [comment]
Apologies: due to time pressures I had to switch to Linux and hence I have no further information on the Windows build.

**Comment by Char-Aznable** [comment]
I've got the same error message on Ubuntu 16.04 with OpenBLAS installed from Linuxbrew. The problem can be get around by setting `-DBLAS_LIBRARIES=$LIB/libopenblas.so`  where `$LIB` is the location of libopenblas.so. But this is just a hack because `BLAS_LIBRARIES` is not supposed to be set by the user but set by cmake instead. This still indicates there's a problem with cmake finding openblas. 

I tried setting `BLA_VENDOR` but it didn't work. 

@martin-frbg

**Comment by martin-frbg** [comment]
Current develop branch has a fix for `find_package(OpenBLAS)` from #1703 that could be related. Apart from that, the behaviour of the cmake FindBLAS macro probably depends on whether your `$LIB` location is one of the regular library search paths of your system (such as /usr/lib, /usr/local/lib etc)
